src/lib/decode.ts

```diff
@@ -0,0 +1,321 @@
+                if (effectIdOld > 15) {
+                    if (verbosity >= 0) {
+                        Log.error(`Movement: Unknown effect id ${effectIdOld}. This is a known bug.`);
+                        console.log('If you know an AVS version that will display this Movement as anything else but "None", then please send it in!');
+                    }
+                    effect = Table.movementEffect[0];
+                } else {
+                    effect = Table.movementEffect[effectIdOld];
+                }
+            }
+        } else {
+            let effectIdNew = 0;
+            if (offset + config.sizeInt * 6 < end) {
+                effectIdNew = get.UInt32(blob, offset + config.sizeInt * 6); // 1*config.sizeInt, because of oldId=0, and 5*config.sizeint because of the other settings.
+            }
+            effect = Table.movementEffect[effectIdNew];
+        }
+        if (effect && effect.length > 0) {
+            comp['builtinEffect'] = effect[0];
+        }
+        comp['output'] = get.UInt32(blob, offset + config.sizeInt) ? '50/50' : 'Replace';
+        comp['sourceMapped'] = get.Bool(blob, offset + config.sizeInt * 2, config.sizeInt)[0];
+        comp['coordinates'] = Table.coordinates[get.UInt32(blob, offset + config.sizeInt * 3)];
+        comp['bilinear'] = get.Bool(blob, offset + config.sizeInt * 4, config.sizeInt)[0];
+        comp['wrap'] = get.Bool(blob, offset + config.sizeInt * 5, config.sizeInt)[0];
+        if (effect && effect.length && effectIdOld !== 1 && effectIdOld !== 7) {
+            // 'slight fuzzify' and 'blocky partial out' have no script representation.
+            code = effect[1];
+            comp['coordinates'] = effect[2]; // overwrite
+        }
+        comp['code'] = code;
+        if (hidden) {
+            comp['_hidden'] = hidden;
+        }
+        return comp;
+    },
+
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    avi(blob: Uint8Array, offset: number): unknown {
+        const comp = {
+            type: 'AVI',
+            group: 'Render',
+            enabled: get.Bool(blob, offset, config.sizeInt)[0]
+        };
+        const strAndSize = get.NtString(blob, offset + config.sizeInt * 3);
+        comp['file'] = strAndSize[0];
+        comp['speed'] = get.UInt32(blob, offset + config.sizeInt * 5 + strAndSize[1]); // 0: fastest, 1000: slowest
+        const beatAdd = get.UInt32(blob, offset + config.sizeInt * 3 + strAndSize[1]);
+        if (beatAdd) {
+            comp['output'] = '50/50';
+        } else {
+            comp['output'] = get.Map8(blob, offset + config.sizeInt, { 0: 'Replace', 1: 'Additive', 0x100000000: '50/50' });
+        }
+        comp['onBeatAdd'] = beatAdd;
+        comp['persist'] = get.UInt32(blob, offset + config.sizeInt * 4 + strAndSize[1]); // 0-32
+
+        return comp;
+    },
+
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    simple(blob: Uint8Array, offset: number): unknown {
```

Comment:
Same here, both `blob` and `offset` look used to me.

Reply:
fixed in 023acc0